### PR TITLE
[FIX] web: test preference on user_menu_items

### PR DESCRIPTION
**Pull request by jpp-odoo**
Test that we are correclty calling correclty orm.call and action.doAction

### CODE REVIEW

File: addons/web/static/tests/webclient/user_menu_tests.js [hunk 1]
```diff
@@ -129,3 +132,46 @@ QUnit.test("display the correct name in debug mode", async (assert) => {
   assert.containsOnce(userMenuEl, "span.oe_topbar_name");
   assert.strictEqual(userMenuEl.querySelector(".oe_topbar_name").textContent, "Sauron (test)");
 });
+
+QUnit.test("can be rendered", async (assert) => {
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/user_menu_tests.js]: copy/paste? :-)

File: addons/web/static/tests/webclient/user_menu_tests.js [hunk 2]
```diff
@@ -129,3 +132,46 @@ QUnit.test("display the correct name in debug mode", async (assert) => {
...
+  const mockRPC = (route) => {
+    if (route === "/web/dataset/call_kw/res.users/action_get") {
+      return Promise.resolve({
+        name: "Change My Preferences",
+        res_id: 0,
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/user_menu_tests.js:141]: I guess the actual call returns res_id: 0?

File: addons/web/static/tests/webclient/user_menu_tests.js [hunk 3]
```diff
@@ -129,3 +132,46 @@ QUnit.test("display the correct name in debug mode", async (assert) => {
...
+  userMenu = await mount(UserMenu, { env, target });
+  let userMenuEl = userMenu.el;
+  await click(userMenu.el.querySelector("button.o_dropdown_toggler"));
+  userMenuEl = userMenu.el;
+  assert.containsOnce(userMenuEl, "ul.o_dropdown_menu li.o_dropdown_item");
+  const children = userMenuEl.querySelector("ul.o_dropdown_menu").firstChild;
+  assert.strictEqual(children.tagName, "LI");
+  const item = userMenuEl.querySelector("ul.o_dropdown_menu li.o_dropdown_item");
+  assert.strictEqual(item.dataset.menu, "settings");
+  const span = userMenuEl.querySelector("ul.o_dropdown_menu li.o_dropdown_item span");
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/user_menu_tests.js]: you can remove the DOM assertions (this test is about the preference item, and its behavior, not the user menu dropdown)

Comment (aab-odoo) [reply addons/web/static/tests/webclient/user_menu_tests.js]: I would just assert that there is one item, and that it's text is "Preferences", then click on it, and verify the steps

File: addons/web/static/tests/webclient/user_menu_tests.js (hunk 2 above)
Comment (jpp-odoo) [reply addons/web/static/tests/webclient/user_menu_tests.js:141]: Indeed.
This is the actual return, I only keep the name and res_id (res_id is needed on the code of preferenceItem)
```
{
    "id": 74,
    "name": "Change My Preferences",
    "type": "ir.actions.act_window",
    "view_id": false,
    "domain": false,
    "context": "{}",
    "res_id": 0,
    "res_model": "res.users",
    "target": "new",
    "view_mode": "form",
    "usage": false,
    "view_ids": [
        14
    ],
    "views": [
        [
            167,
            "form"
        ]
    ],
    "limit": 80,
    "groups_id": [],
    "search_view_id": false,
    "filter": false,
    "search_view": "{'model': 'res.users', 'field_parent': False, 'arch': '<search string=\"Users\">\\n                    <field name=\"name\" filter_domain=\"[\\'|\\', \\'|\\', (\\'name\\',\\'ilike\\',self), (\\'login\\',\\'ilike\\',self), (\\'email\\',\\'ilike\\',self)]\" string=\"User\" modifiers=\"{}\"/>\\n                    <field name=\"company_ids\" string=\"Company\" can_create=\"true\" can_write=\"true\" invisible=\"1\" modifiers=\"{&quot;invisible&quot;: true}\"/>\\n                    <field name=\"share\" modifiers=\"{&quot;readonly&quot;: true}\"/>\\n                    <filter name=\"no_share\" string=\"Internal Users\" domain=\"[(\\'share\\',\\'=\\',False)]\"/>\\n                    <separator/>\\n                    <filter name=\"Inactive\" string=\"Inactive Users\" domain=\"[(\\'active\\',\\'=\\',False)]\"/>\\n                </search>', 'name': 'res.users.search', 'type': 'search', 'view_id': 163, 'base_model': 'res.users', 'fields': {'name': {'id': None, 'select': None, 'views': {}, 'type': 'char', 'change_default': False, 'company_dependent': False, 'depends': ('partner_id.name',), 'manual': False, 'readonly': False, 'related': ('partner_id', 'name'), 'required': False, 'searchable': True, 'sortable': True, 'store': False, 'string': 'Name', 'translate': False, 'trim': True}, 'company_ids': {'id': None, 'select': None, 'views': {}, 'type': 'many2many', 'change_default': False, 'company_dependent': False, 'context': {}, 'depends': (), 'domain': [], 'manual': False, 'readonly': False, 'relation': 'res.company', 'required': False, 'searchable': True, 'sortable': False, 'store': True, 'string': 'Companies'}, 'share': {'id': None, 'select': None, 'views': {}, 'type': 'boolean', 'change_default': False, 'company_dependent': False, 'depends': ('groups_id',), 'help': 'External user with limited access, created only for the purpose of sharing data.', 'manual': False, 'readonly': True, 'required': False, 'searchable': True, 'sortable': True, 'store': True, 'string': 'Share User'}}}",
    "xml_id": "base.action_res_users_my",
    "help": false,
    "binding_model_id": false,
    "binding_type": "action",
    "binding_view_types": "list,form",
    "display_name": "Change My Preferences",
    "create_uid": [
        1,
        "System"
    ],
    "create_date": "2021-05-03 07:20:56",
    "write_uid": [
        1,
        "System"
    ],
    "write_date": "2021-05-03 07:20:56",
    "__last_update": "2021-05-03 07:20:56"
}
```

Comment (aab-odoo) [reply addons/web/static/tests/webclient/user_menu_tests.js:141]: ok

File: addons/web/static/tests/webclient/user_menu_tests.js [hunk 4]
```diff
@@ -129,3 +132,42 @@ QUnit.test("display the correct name in debug mode", async (assert) => {
...
+        },
+      };
+    },
+  };
+  serviceRegistry.add("action", fakeActionService, { force: true });
+
+  env = await makeTestEnv(testConfig);
+  userMenuRegistry.add("profile", preferencesItem);
+  userMenu = await mount(UserMenu, { env, target });
+  let userMenuEl = userMenu.el;
```
Comment (aab-odoo) [inline addons/web/static/tests/webclient/user_menu_tests.js]: userMenuEl not really necessary, you can remove l 165 and l 167 and use userMenu.el everywhere

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**